brianzinn/react-babylonjs
# Static site 👍

Hey @brianzinn ok this is a big update and I'm feeling really great about this path. We're using gatsby and dumping storybook. 

Features:
* Custom React component (dev tool) that can show code and output, in both TS and JS, along with ability to launch a TS-based or JS-based sandbox
* 1st class Typescript support with automatic Javascript transpiling. We maintain everything in TS, it transpiles to JS
* Custom webpack loader to detect changes in demo files and rebuild when in dev mode

All that functionality lives in the `/devtools` folder, while `/packages` contains Babylon-specific stuff. I would like it very much if we could incubate the `@devtools` packages in this repo until we feel like they are robust enough to put into a separate monorepo and publish on npm. Thanks for this opportunity to make something really nice!

To test:

```bash
rm -rf node_modules
yarn
yarn workspaces build # this will build all packages in the correct order
cd packages/static
rm -rf .cache && yarn start
```

Then go to `packages/static/content` and edit the `mdx` files and supporting `tsx` files to your heart's content. Every `.tsx` demo must be free-standing, and it is included in the mdx using:

`[devtool:MyDemo.tsx]` which should reside in the same folder.

```
content/myPage
     index.mdx
     MyDemo.tsx
```

To develop:

Go into any `/devtools` package and do `yarn build && yarn watch`. That will pick up changes and rebuild the js `dist` folders. You may need to restart the static site depending on the changes.
